### ISSUE TITLE
Uppercase Headers

### ISSUE THREAD

**Issue by evanleck** [ohler55/agoo#21]
I am trying out Agoo (it's really dang fast!) with an existing Rack project that's been running behind Puma and noticed two things about the headers that caught my eye.

1. Though it may not be formally documented anywhere, I believe all uppercase header keys should be snake-case. For example, I believe `HTTP_USER-AGENT` should be `HTTP_USER_AGENT` (not the underscore instead of the dash).
2. The keys alone should be uppercase, not the values.

**Comment by evanleck** [comment]
Yeah, the spec doesn't mention it at all but it seems to be the convention that Rack and Rails follow, at least looking through some tests like these:

* https://github.com/rails/rails/blob/master/actionpack/test/dispatch/request_test.rb#L511
* https://github.com/rack/rack/blob/master/test/spec_request.rb#L126
* https://github.com/rack/rack/blob/master/test/spec_request.rb#L484

That's awesome, I'm excited to try it out!

**Comment by ohler55** [comment]
Thanks for the references. Very helpful. Anyway, I'll change tonight and push to the 'cluster' branch which is ready for release after a bit more testing.

**Comment by evanleck** [comment]
Awesome, thank you!

**Comment by ohler55** [comment]
Released with the fix and with clustering.

**Comment by evanleck** [comment]
That looks a lot better as far as the keys go, but it looks like the header parsing is incorrect. I'm _guessing_ that the headers are being split on the last occurrence of a colon? Check out the lines starting with `HTTP_USER_AGENT` and `HTTP_HOST` below. I'm running on `localhost:3000` and the user-agent sent by the browser is `Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv:60.0) Gecko/20100101 Firefox/60.0`.

```
"REQUEST_METHOD"=>"GET",
"SCRIPT_NAME"=>"",
"PATH_INFO"=>"/logout",
"QUERY_STRING"=>"",
"SERVER_NAME"=>"localhost",
"SERVER_PORT"=>"3000",
"HTTP_Host: localhost"=>"3000",
"HTTP_HOST: LOCALHOST"=>"3000",
"HTTP_User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.13; rv"=> "60.0) Gecko/20100101 Firefox/60.0",
"HTTP_USER_AGENT: MOZILLA/5.0 (MACINTOSH; INTEL MAC OS X 10.13; RV"=> "60.0) Gecko/20100101 Firefox/60.0",
"HTTP_Accept"=> "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
"HTTP_ACCEPT"=> "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
"HTTP_Accept-Language"=>"en-US,en;q=0.5",
"HTTP_ACCEPT_LANGUAGE"=>"en-US,en;q=0.5",
"HTTP_Accept-Encoding"=>"gzip, deflate",
"HTTP_ACCEPT_ENCODING"=>"gzip, deflate"
```

`HTTP_USER_AGENT` isn't even a key in the hash, the whole key is `HTTP_USER_AGENT: MOZILLA/5.0 (MACINTOSH; INTEL MAC OS X 10.13; RV` without a value. Where the `HTTP_HOST` header should be is the key `HTTP_HOST: LOCALHOST` with the value `3000`.

**Comment by ohler55** [comment]
I'll fix that tonight. Thanks.

**Comment by evanleck** [comment]
Fixed with b68b08972491af4042be1d324bb47de1848c382b, thanks!
